util_test.go

```diff
@@ -4,11 +4,29 @@
 package raft
 
 import (
+	"bytes"
 	"regexp"
 	"testing"
 	"time"
 )
 
+// TestMsgpackEncodeTime ensures that we don't break backwards compatibility when updating go-msgpack with
+// Raft binary formats.
+func TestMsgpackEncodeTimeDefaultFormat(t *testing.T) {
+	stamp := "2006-01-02T15:04:05Z"
+	tm, err := time.Parse(time.RFC3339, stamp)
+	if err != nil {
+		t.Fatal(err)
+	}
+	buf, err := encodeMsgPack(tm)
+
+	expected := []byte{175, 1, 0, 0, 0, 14, 187, 75, 55, 229, 0, 0, 0, 0, 255, 255}
+
+	if !bytes.Equal(buf.Bytes(), expected) {
+		t.Errorf("Expected time %s to encode as %+v but got %+v", stamp, expected, buf.Bytes())
+	}
+}
+
```

Comment:
Would it be worth having a similar test that encodes an entire `Log` populated with some arbitrary data? My thinking is that that struct is really the main thing msgpack is used for other than some minor RPC framing.

As far as I know the only part of the encoding that caused issues due to changes upstream was the time.Time in the `AppendedAt` field which this covers, but it might provided added confidence that there are no other unknown incompatibilities too?

Not essential, just a thought.

Reply:
So, there are some tests like that in `go-msgpack` already: https://github.com/hashicorp/go-msgpack/blob/main/codec/raft_test.go which links to a bunch of data I captured from running Raft with various versions of `go-msgpack` https://github.com/hashicorp/go-msgpack/tree/main/codec/internal/testdata